nsqd/protocol_v2.go

```diff
@@ -504,7 +504,7 @@ func (p *protocolV2) AUTH(client *clientV2, params [][]byte) ([]byte, error) {
 		return nil, protocol.NewFatalClientErr(err, "E_BAD_BODY", "AUTH failed to read body")
 	}
 
-	if client.HasAuthorizations() {
+	if client.ClientAuth.HasAuthorizations() {
```

Comment:
Related, I think this is an indication that we might want to _embed_ `ClientAuth` in `clientV2` so that we don't have to change the way we access these methods (feels redundant to write `client.ClientAuth.<blah>`).